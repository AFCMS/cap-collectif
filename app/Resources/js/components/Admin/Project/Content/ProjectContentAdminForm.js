// @flow
import React from 'react';
import { connect } from 'react-redux';
import { Field } from 'redux-form';
import { createFragmentContainer, graphql } from 'react-relay';
import { type IntlShape, FormattedMessage } from 'react-intl';

import renderComponent from '../../../Form/Field';
import UserListField from '../../Field/UserListField';
import { type Author } from '../Form/ProjectAdminForm';
import ProjectTypeListField from '../../Field/ProjectTypeListField';
import { type ProjectContentAdminForm_project } from '~relay/ProjectContentAdminForm_project.graphql';

export type FormValues = {|
  title: string,
  authors: Author[],
  opinionTerm: number,
  projectType: string,
|};

type Props = {|
  ...ReduxFormFormProps,
  project: ?ProjectContentAdminForm_project,
  intl: IntlShape,
  formName: string,
|};

const opinionTerms = [
  {
    id: 0,
    label: 'project.opinion_term.opinion',
  },
  {
    id: 1,
    label: 'project.opinion_term.article',
  },
];

const convertOpinionTerm = (opinionTerm: string): number => parseInt(opinionTerm, 10);

const formName = 'projectAdminForm';

export const ProjectContentAdminForm = (props: Props) => {
  const { handleSubmit, intl } = props;

  return (
    <div className="col-md-12">
      <div className="box box-primary container-fluid">
        <div className="box-header">
          <h4 className="box-title">
            <FormattedMessage id="admin.group.content" />
          </h4>
        </div>
        <div className="box-content">
          <form onSubmit={handleSubmit} id={formName}>
            <Field
              type="text"
              name="title"
              label={
                <div>
                  <FormattedMessage id="admin.fields.group.title" />
                  <span className="excerpt">
                    <FormattedMessage id="global.mandatory" />
                  </span>
                </div>
              }
              component={renderComponent}
            />
            <UserListField
              id="project-author"
              name="authors"
              clearable
              selectFieldIsObject
              debounce
              autoload={false}
              multi
              labelClassName="control-label"
              inputClassName="fake-inputClassName"
              placeholder={intl.formatMessage({ id: 'all-the-authors' })}
              label={
                <div>
                  <FormattedMessage id="admin.fields.project.authors" />
                  <span className="excerpt">
                    <FormattedMessage id="global.mandatory" />
                  </span>
                </div>
              }
              ariaControls="EventListFilters-filter-author-listbox"
            />

            <ProjectTypeListField />
            <Field
              name="opinionTerm"
              type="select"
              component={renderComponent}
              parse={convertOpinionTerm}
              normalize={convertOpinionTerm}
              label={
                <span>
                  <FormattedMessage id="admin.fields.project.opinion_term" />
                </span>
              }>
              {opinionTerms.map(projectTerm => (
                <option key={projectTerm.id} value={projectTerm.id}>
                  {intl.formatMessage({ id: projectTerm.label })}
                </option>
              ))}
            </Field>
          </form>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state, { project }: Props) => ({
  initialValues: {
    opinionTerm: project ? project.opinionTerm : opinionTerms[0].id,
    authors: project ? project.authors : [],
    title: project ? project.title : null,
    projectType: project && project.type ? project.type.id : null,
  },
});

export const container = connect(mapStateToProps)(ProjectContentAdminForm);

export default createFragmentContainer(container, {
  project: graphql`
    fragment ProjectContentAdminForm_project on Project {
      id
      title
      authors {
        value: id
        label: username
      }
      opinionTerm
      type {
        id
      }
    }
  `,
});
