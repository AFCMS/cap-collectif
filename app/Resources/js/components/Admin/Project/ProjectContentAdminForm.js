// @flow
import React from 'react';
import { connect } from 'react-redux';
import { Button } from 'react-bootstrap';
import { type FormProps, reduxForm, Field } from 'redux-form';
import { createFragmentContainer, graphql } from 'react-relay';
import { injectIntl, type IntlShape, FormattedMessage } from 'react-intl';

import AlertForm from '../../Alert/AlertForm';
import renderComponent from '../../Form/Field';
import type { Dispatch } from '../../../types';
import UserListField from '../Field/UserListField';
import AppDispatcher from '../../../dispatchers/AppDispatcher';
import ProjectTypeListField from '../Field/ProjectTypeListField';
import { UPDATE_ALERT } from '../../../constants/AlertConstants';
import CreateProjectMutation from '../../../mutations/CreateProjectMutation';
import UpdateProjectMutation from '../../../mutations/UpdateProjectMutation';
import { type ProjectContentAdminForm_project } from '~relay/ProjectContentAdminForm_project.graphql';

type Props = {|
  ...FormProps,
  project: ?ProjectContentAdminForm_project,
  handleSubmit: () => void,
  intl: IntlShape,
  formName: string,
|};

type Author = {|
  value: string,
  label: string,
|};

type FormValues = {|
  title: string,
  authors: Author[],
  opinionTerm: string,
  projectType: string,
|};

const projectTerms = [
  {
    id: 0,
    label: 'project.opinion_term.opinion',
  },
  {
    id: 1,
    label: 'project.opinion_term.article',
  },
];

const formatAuthors = (authors: Author[]): string[] => authors.map(author => author.value);

const onSubmit = (
  { title, authors, opinionTerm, projectType }: FormValues,
  dispatch: Dispatch,
  props: Props,
) => {
  const input = {
    title,
    opinionTerm,
    projectType,
    authors: formatAuthors(authors),
  };
  if (props.project) {
    return UpdateProjectMutation.commit({
      input: {
        id: props.project.id,
        ...input,
      },
    }).then(data => {
      if (data.updateProject && data.updateProject.project) {
        AppDispatcher.dispatch({
          actionType: UPDATE_ALERT,
          alert: { bsStyle: 'success', content: 'alert.success.report.argument' },
        });
      }
    });
  }
  return CreateProjectMutation.commit({ input }).then(data => {
    if (data.createProject && data.createProject.project) {
      window.location.href = data.createProject.project.adminUrl;
      AppDispatcher.dispatch({
        actionType: UPDATE_ALERT,
        alert: { bsStyle: 'success', content: 'alert.success.report.argument' },
      });
    }
  });
};

const validate = ({ title, authors, opinionTerm, projectType }: FormValues) => {
  const errors = {};

  if (!title || title.length < 2) {
    errors.title = 'global.required';
  }

  if (!authors || authors.length <= 0) {
    errors.authors = 'global.required';
  }

  if (!projectType) {
    errors.projectType = 'global.required';
  }

  if (!opinionTerm) {
    errors.opinionTerm = 'global.required';
  }

  return errors;
};

const formName = 'projectAdminForm';

export const ProjectContentAdminForm = (props: Props) => {
  const {
    handleSubmit,
    intl,
    valid,
    invalid,
    submitting,
    pristine,
    submitSucceeded,
    submitFailed,
  } = props;

  return (
    <form onSubmit={handleSubmit} id={formName}>
      <Field
        type="text"
        name="title"
        label={<FormattedMessage id="admin.fields.group.title" />}
        component={renderComponent}
      />
      <UserListField
        id="project-author"
        name="authors"
        clearable
        selectFieldIsObject
        debounce
        autoload={false}
        multi
        labelClassName="control-label"
        inputClassName="fake-inputClassName"
        placeholder={intl.formatMessage({ id: 'all-the-authors' })}
        label={intl.formatMessage({ id: 'admin.fields.project.authors' })}
        ariaControls="EventListFilters-filter-author-listbox"
      />

      <ProjectTypeListField />
      <Field
        name="opinionTerm"
        type="select"
        component={renderComponent}
        initialValue={projectTerms[0].id}
        label={
          <span>
            <FormattedMessage id="admin.fields.project.opinion_term" />
          </span>
        }>
        {projectTerms.map(projectTerm => (
          <option key={projectTerm.id} value={projectTerm.id}>
            {intl.formatMessage({ id: projectTerm.label })}
          </option>
        ))}
      </Field>
      <Button
        id="submit-project-content"
        type="submit"
        disabled={invalid || submitting || pristine}
        bsStyle="primary">
        {submitting ? (
          <FormattedMessage id="global.loading" />
        ) : (
          <FormattedMessage id="global.save" />
        )}
      </Button>
      <AlertForm
        valid={valid}
        invalid={invalid && !pristine}
        submitSucceeded={submitSucceeded}
        submitFailed={submitFailed}
        submitting={submitting}
      />
    </form>
  );
};

const mapStateToProps = (state, { project }: Props) => ({
  initialValues: {
    opinionTerm: project ? project.opinionTerm : projectTerms[0].id,
    authors: project ? project.authors : [],
    title: project ? project.title : null,
    projectType: project && project.type ? project.type.id : null,
  },
});

const form = injectIntl(
  reduxForm({
    validate,
    onSubmit,
    form: formName,
  })(ProjectContentAdminForm),
);

export const container = connect(mapStateToProps)(form);

export default createFragmentContainer(container, {
  project: graphql`
    fragment ProjectContentAdminForm_project on Project {
      id
      title
      authors {
        value: id
        label: username
      }
      opinionTerm
      type {
        id
      }
    }
  `,
});
