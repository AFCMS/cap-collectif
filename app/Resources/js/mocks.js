// @flow
/* eslint-env jest */
import { type FormProps } from 'redux-form';
import { type IntlShape } from 'react-intl';
import type { RelayPaginationProp, RelayRefetchProp } from 'react-relay';

export const $refType: any = null;
export const $fragmentRefs: any = null;

export const intlMock: IntlShape = {
  locale: 'fr-FR',
  formats: {},
  messages: {},
  now: () => 0,
  formatHTMLMessage: message => String(message),
  formatPlural: message => String(message),
  formatNumber: message => String(message),
  formatRelative: message => String(message),
  formatTime: message => String(message),
  formatDate: message => String(message),
  formatMessage: message => String(message.id),
};

export const formMock: FormProps = {
  anyTouched: false,
  array: {
    insert: jest.fn(),
    move: jest.fn(),
    pop: jest.fn(),
    push: jest.fn(),
    remove: jest.fn(),
    removeAll: jest.fn(),
    shift: jest.fn(),
    splice: jest.fn(),
    swap: jest.fn(),
    unshift: jest.fn(),
  },
  asyncValidate: jest.fn(),
  asyncValidating: false,
  autofill: jest.fn(),
  blur: jest.fn(),
  change: jest.fn(),
  clearAsyncError: jest.fn(),
  clearSubmit: jest.fn(),
  destroy: jest.fn(),
  dirty: false,
  dispatch: jest.fn(),
  error: null,
  form: 'formName',
  handleSubmit: jest.fn(),
  initialize: jest.fn(),
  initialized: true,
  initialValues: {},
  invalid: false,
  pristine: true,
  reset: jest.fn(),
  submitting: false,
  submitFailed: false,
  submitSucceeded: false,
  touch: jest.fn(),
  untouch: jest.fn(),
  valid: true,
  warning: null,
};

export const relayPaginationMock: RelayPaginationProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  hasMore: () => false,
  isLoading: () => false,
  loadMore: jest.fn(),
  refetchConnection: jest.fn(),
};

export const relayRefetchMock: RelayRefetchProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  refetch: jest.fn(),
};
