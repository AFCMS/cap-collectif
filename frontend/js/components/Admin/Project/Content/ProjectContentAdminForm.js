// @flow
import React, { type ComponentType } from 'react';
import { Field } from 'redux-form';
import styled, { type StyledComponent } from 'styled-components';
import { OverlayTrigger } from 'react-bootstrap';
import { createFragmentContainer, graphql } from 'react-relay';
import { FormattedHTMLMessage, FormattedMessage, type IntlShape } from 'react-intl';
import select from '~/components/Form/Select';
import renderComponent from '~/components/Form/Field';
import UserListField from '../../Field/UserListField';
import { type Author } from '../Form/ProjectAdminForm';
import ProjectTypeListField from '../../Field/ProjectTypeListField';
import { type ProjectContentAdminForm_project } from '~relay/ProjectContentAdminForm_project.graphql';
import { loadDistrictOptions, loadThemeOptions } from '../Metadata/ProjectMetadataAdminForm';
import Tooltip from '~/components/Utils/Tooltip';
import {
  ProjectBoxHeader,
  ProjectSmallInput,
  ProjectBoxContainer,
} from '../Form/ProjectAdminForm.style';
import Icon, { ICON_NAME } from '~ui/Icons/Icon';

type Option = {|
  value: string,
  label: string,
|};

export type FormValues = {|
  title: string,
  authors: Author[],
  projectType: string,
  themes: Option[],
  Cover: ?{
    id: string,
    description: ?string,
    name: ?string,
    size: ?string,
    url: ?string,
  },
  metaDescription: ?string,
  video: ?string,
  districts: Option[],
|};

type Props = {|
  ...ReduxFormFormProps,
  project: ?ProjectContentAdminForm_project,
  intl: IntlShape,
|};

export const InformationIcon: StyledComponent<{}, {}, ComponentType<any>> = styled(Icon).attrs({
  name: ICON_NAME.information,
  size: '12px',
  style: undefined, // Override default styles that `Icon` component apply
})`
  opacity: 0.3;
  position: relative;
  top: 1px;
`;

export const renderLabel = (id: string, intl: IntlShape, helpText?: string, optional?: boolean) => (
  <div>
    {intl.formatMessage({ id })}
    <span className="excerpt inline">
      {!optional && intl.formatMessage({ id: 'global.optional' })}{' '}
      {helpText && (
        <OverlayTrigger
          key="top"
          placement="top"
          overlay={
            <Tooltip id="tooltip-top" className="text-left" style={{ wordBreak: 'break-word' }}>
              <FormattedHTMLMessage id={helpText} />
            </Tooltip>
          }>
          <InformationIcon />
        </OverlayTrigger>
      )}
    </span>
  </div>
);

export const validate = (props: FormValues) => {
  const { title, authors } = props;
  const errors = {};

  if (!title || title.length < 2) {
    errors.title = 'global.required';
  }

  if (!authors || authors.length <= 0) {
    errors.authors = 'global.required';
  }

  return errors;
};

export const ProjectContentAdminForm = ({ intl }: Props) => {
  return (
    <div className="col-md-12">
      <ProjectBoxContainer className="box container-fluid">
        <ProjectBoxHeader>
          <h4>
            <FormattedMessage id="global.general" />
          </h4>
        </ProjectBoxHeader>
        <div className="box-content">
          <Field
            type="text"
            name="title"
            label={<FormattedMessage id="global.title" />}
            component={renderComponent}
          />
          <UserListField
            id="project-author"
            name="authors"
            clearable
            selectFieldIsObject
            debounce
            autoload={false}
            multi
            placeholder=" "
            labelClassName="control-label"
            inputClassName="fake-inputClassName"
            label={<FormattedMessage id="admin.fields.project.authors" />}
            ariaControls="EventListFilters-filter-author-listbox"
          />

          <ProjectSmallInput>
            <ProjectTypeListField optional placeholder=" " />
          </ProjectSmallInput>
          <div className="row mr-0 ml-0">
            <Field
              id="cover"
              name="Cover"
              component={renderComponent}
              type="image"
              label={renderLabel('proposal.media', intl)}
            />
          </div>
          <Field
            type="text"
            name="video"
            id="video"
            label={renderLabel('admin.fields.project.video', intl, 'admin.help.project.video')}
            placeholder={intl.formatMessage({ id: 'admin-project-video-placeholder' })}
            component={renderComponent}
          />
          <Field
            selectFieldIsObject
            debounce
            autoload
            labelClassName="control-label"
            inputClassName="fake-inputClassName"
            component={select}
            id="themes"
            name="themes"
            placeholder=" "
            label={renderLabel('global.themes', intl)}
            role="combobox"
            aria-autocomplete="list"
            aria-haspopup="true"
            loadOptions={loadThemeOptions}
            multi
            clearable
          />
          <Field
            role="combobox"
            aria-autocomplete="list"
            aria-haspopup="true"
            loadOptions={loadDistrictOptions}
            component={select}
            id="districts"
            name="districts"
            clearable
            selectFieldIsObject
            debounce
            autoload
            multi
            labelClassName="control-label"
            inputClassName="fake-inputClassName"
            label={renderLabel('proposal_form.districts', intl)}
          />
          <Field
            name="metaDescription"
            type="textarea"
            label={renderLabel('global.meta.description', intl, 'admin.help.metadescription')}
            component={renderComponent}
          />
        </div>
      </ProjectBoxContainer>
    </div>
  );
};

export default createFragmentContainer(ProjectContentAdminForm, {
  project: graphql`
    fragment ProjectContentAdminForm_project on Project {
      id
      title
      metaDescription
      authors {
        value: id
        label: username
      }
      type {
        id
      }
      Cover: cover {
        id
        name
        size
        url
      }
      video
      themes {
        value: id
        label: title
      }
      districts {
        edges {
          node {
            value: id
            label: name
          }
        }
      }
    }
  `,
});
