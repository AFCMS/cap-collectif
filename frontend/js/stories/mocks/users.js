// @flow

export const author = {
  id: 'user1',
  username: 'admin',
  media: {
    url: 'https://source.unsplash.com/random/150x150',
  },
  _links: {},
  vip: false,
  isViewer: false,
  isAdmin: true,
  isEvaluer: true,
};

export const followers = [
  {
    displayName: 'All Fred',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'guemas',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 0,
  },
  {
    displayName: 'Shezeboss',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'AS',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 240,
  },
  {
    displayName: 'Louisette',
    isEmailConfirmed: false,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 1,
  },
  {
    contributionsCount: 1,
    isEmailConfirmed: true,
    _links: null,
    media: null,
  },
  {
    displayName: 'guemas',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 0,
  },
  {
    displayName: 'Shezeboss',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'AS',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 240,
  },
];
