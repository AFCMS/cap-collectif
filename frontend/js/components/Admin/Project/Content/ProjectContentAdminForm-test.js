// @flow
/* eslint-env jest */
import * as React from 'react';
import { shallow } from 'enzyme';
import { ProjectContentAdminForm } from './ProjectContentAdminForm';
import { formMock, intlMock, $refType } from '~/mocks';
import MockProviders from '~/testUtils';

describe('<ProjectContentAdminForm />', () => {
  const defaultProps = {
    ...formMock,
    intl: intlMock,
    project: null,
  };

  it('renders correctly empty', () => {
    const wrapper = shallow(
      <MockProviders store={{ user: { user: { isAdmin: false } } }}>
        <ProjectContentAdminForm {...defaultProps} />
      </MockProviders>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with a project', () => {
    const props = {
      ...defaultProps,
      project: {
        $refType,
        id: '1',
        title: 'testTitle',
        type: {
          id: '1',
        },
        video: 'dailymotion.com/issou',
        Cover: null,
        authors: [],
        themes: [],
        districts: null,
        metaDescription: 'so meta',
      },
    };
    const wrapper = shallow(
      <MockProviders store={{ user: { user: { isAdmin: false } } }}>
        <ProjectContentAdminForm {...props} />
      </MockProviders>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
