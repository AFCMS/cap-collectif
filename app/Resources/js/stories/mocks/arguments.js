// @flow
import { author } from './users';

export const argument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const argumentsData = [
  { ...argument, user: { ...argument.user, vip: true } },
  { ...argument },
  { ...argument, trashedStatus: 'INVISIBLE' },
  { ...argument },
];
