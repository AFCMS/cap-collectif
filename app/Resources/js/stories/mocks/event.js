// @flow
import { author } from './users';

export const event = {
  title: 'PHPTourDuTurfu',
  url: '#',
  image: {
    url: 'https://picsum.photos/300/400',
    alt: 'event picture',
  },
  date: {
    start: '2030-03-10 00:00:00',
    end: '2032-06-15 00:00:00',
  },
  location: {
    place: 'Tour Eiffel',
    zip: '75007',
    city: 'Paris',
  },
  category: 'Immobilier',

  user: author,
};
