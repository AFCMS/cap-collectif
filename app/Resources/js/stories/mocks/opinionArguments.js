// @flow
import { author } from './users';

export const opinionArgument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const opinionArguments = [
  { ...opinionArgument, user: { ...opinionArgument.user, vip: true } },
  { ...opinionArgument },
  { ...opinionArgument, trashedStatus: 'INVISIBLE' },
  { ...opinionArgument },
];
