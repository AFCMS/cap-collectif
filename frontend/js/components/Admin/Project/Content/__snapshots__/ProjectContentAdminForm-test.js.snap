// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectContentAdminForm /> renders correctly empty 1`] = `
<div
  className="col-md-12"
>
  <ProjectAdminFormstyle__ProjectBoxContainer
    className="box container-fluid"
  >
    <ProjectAdminFormstyle__ProjectBoxHeader>
      <h4>
        <FormattedMessage
          id="global.general"
          values={Object {}}
        />
      </h4>
    </ProjectAdminFormstyle__ProjectBoxHeader>
    <div
      className="box-content"
    >
      <Field
        component={[Function]}
        label={
          <FormattedMessage
            id="global.title"
            values={Object {}}
          />
        }
        name="title"
        type="text"
      />
      <UserListField
        ariaControls="EventListFilters-filter-author-listbox"
        authorOfEvent={false}
        autoload={false}
        className=""
        clearable={true}
        debounce={true}
        id="project-author"
        inputClassName="fake-inputClassName"
        label={
          <FormattedMessage
            id="admin.fields.project.authors"
            values={Object {}}
          />
        }
        labelClassName="control-label"
        multi={true}
        name="authors"
        placeholder=" "
        selectFieldIsObject={true}
      />
      <ProjectAdminFormstyle__ProjectSmallInput>
        <injectIntl(ProjectTypeListField)
          optional={true}
          placeholder=" "
        />
      </ProjectAdminFormstyle__ProjectSmallInput>
      <div
        className="row mr-0 ml-0"
      >
        <Field
          component={[Function]}
          id="cover"
          label={
            <div>
              proposal.media
              <span
                className="excerpt inline"
              >
                global.optional
                 
              </span>
            </div>
          }
          name="Cover"
          type="image"
        />
      </div>
      <Field
        component={[Function]}
        id="video"
        label={
          <div>
            admin.fields.project.video
            <span
              className="excerpt inline"
            >
              global.optional
               
              <OverlayTrigger
                defaultOverlayShown={false}
                overlay={
                  <_default
                    className="text-left"
                    id="tooltip-top"
                    style={
                      Object {
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    <FormattedHTMLMessage
                      id="admin.help.project.video"
                      tagName="span"
                      values={Object {}}
                    />
                  </_default>
                }
                placement="top"
                trigger={
                  Array [
                    "hover",
                    "focus",
                  ]
                }
              >
                <ForwardRef(ProjectContentAdminForm__InformationIcon) />
              </OverlayTrigger>
            </span>
          </div>
        }
        name="video"
        placeholder="admin-project-video-placeholder"
        type="text"
      />
      <Field
        aria-autocomplete="list"
        aria-haspopup="true"
        autoload={true}
        clearable={true}
        component={[Function]}
        debounce={true}
        id="themes"
        inputClassName="fake-inputClassName"
        label={
          <div>
            global.themes
            <span
              className="excerpt inline"
            >
              global.optional
               
            </span>
          </div>
        }
        labelClassName="control-label"
        loadOptions={[Function]}
        multi={true}
        name="themes"
        placeholder=" "
        role="combobox"
        selectFieldIsObject={true}
      />
      <Field
        aria-autocomplete="list"
        aria-haspopup="true"
        autoload={true}
        clearable={true}
        component={[Function]}
        debounce={true}
        id="districts"
        inputClassName="fake-inputClassName"
        label={
          <div>
            proposal_form.districts
            <span
              className="excerpt inline"
            >
              global.optional
               
            </span>
          </div>
        }
        labelClassName="control-label"
        loadOptions={[Function]}
        multi={true}
        name="districts"
        placeholder=" "
        role="combobox"
        selectFieldIsObject={true}
      />
      <Field
        component={[Function]}
        label={
          <div>
            global.meta.description
            <span
              className="excerpt inline"
            >
              global.optional
               
              <OverlayTrigger
                defaultOverlayShown={false}
                overlay={
                  <_default
                    className="text-left"
                    id="tooltip-top"
                    style={
                      Object {
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    <FormattedHTMLMessage
                      id="admin.help.metadescription"
                      tagName="span"
                      values={Object {}}
                    />
                  </_default>
                }
                placement="top"
                trigger={
                  Array [
                    "hover",
                    "focus",
                  ]
                }
              >
                <ForwardRef(ProjectContentAdminForm__InformationIcon) />
              </OverlayTrigger>
            </span>
          </div>
        }
        name="metaDescription"
        type="textarea"
      />
    </div>
  </ProjectAdminFormstyle__ProjectBoxContainer>
</div>
`;

exports[`<ProjectContentAdminForm /> renders correctly with a project 1`] = `
<div
  className="col-md-12"
>
  <ProjectAdminFormstyle__ProjectBoxContainer
    className="box container-fluid"
  >
    <ProjectAdminFormstyle__ProjectBoxHeader>
      <h4>
        <FormattedMessage
          id="global.general"
          values={Object {}}
        />
      </h4>
    </ProjectAdminFormstyle__ProjectBoxHeader>
    <div
      className="box-content"
    >
      <Field
        component={[Function]}
        label={
          <FormattedMessage
            id="global.title"
            values={Object {}}
          />
        }
        name="title"
        type="text"
      />
      <UserListField
        ariaControls="EventListFilters-filter-author-listbox"
        authorOfEvent={false}
        autoload={false}
        className=""
        clearable={true}
        debounce={true}
        id="project-author"
        inputClassName="fake-inputClassName"
        label={
          <FormattedMessage
            id="admin.fields.project.authors"
            values={Object {}}
          />
        }
        labelClassName="control-label"
        multi={true}
        name="authors"
        placeholder=" "
        selectFieldIsObject={true}
      />
      <ProjectAdminFormstyle__ProjectSmallInput>
        <injectIntl(ProjectTypeListField)
          optional={true}
          placeholder=" "
        />
      </ProjectAdminFormstyle__ProjectSmallInput>
      <div
        className="row mr-0 ml-0"
      >
        <Field
          component={[Function]}
          id="cover"
          label={
            <div>
              proposal.media
              <span
                className="excerpt inline"
              >
                global.optional
                 
              </span>
            </div>
          }
          name="Cover"
          type="image"
        />
      </div>
      <Field
        component={[Function]}
        id="video"
        label={
          <div>
            admin.fields.project.video
            <span
              className="excerpt inline"
            >
              global.optional
               
              <OverlayTrigger
                defaultOverlayShown={false}
                overlay={
                  <_default
                    className="text-left"
                    id="tooltip-top"
                    style={
                      Object {
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    <FormattedHTMLMessage
                      id="admin.help.project.video"
                      tagName="span"
                      values={Object {}}
                    />
                  </_default>
                }
                placement="top"
                trigger={
                  Array [
                    "hover",
                    "focus",
                  ]
                }
              >
                <ForwardRef(ProjectContentAdminForm__InformationIcon) />
              </OverlayTrigger>
            </span>
          </div>
        }
        name="video"
        placeholder="admin-project-video-placeholder"
        type="text"
      />
      <Field
        aria-autocomplete="list"
        aria-haspopup="true"
        autoload={true}
        clearable={true}
        component={[Function]}
        debounce={true}
        id="themes"
        inputClassName="fake-inputClassName"
        label={
          <div>
            global.themes
            <span
              className="excerpt inline"
            >
              global.optional
               
            </span>
          </div>
        }
        labelClassName="control-label"
        loadOptions={[Function]}
        multi={true}
        name="themes"
        placeholder=" "
        role="combobox"
        selectFieldIsObject={true}
      />
      <Field
        aria-autocomplete="list"
        aria-haspopup="true"
        autoload={true}
        clearable={true}
        component={[Function]}
        debounce={true}
        id="districts"
        inputClassName="fake-inputClassName"
        label={
          <div>
            proposal_form.districts
            <span
              className="excerpt inline"
            >
              global.optional
               
            </span>
          </div>
        }
        labelClassName="control-label"
        loadOptions={[Function]}
        multi={true}
        name="districts"
        placeholder=" "
        role="combobox"
        selectFieldIsObject={true}
      />
      <Field
        component={[Function]}
        label={
          <div>
            global.meta.description
            <span
              className="excerpt inline"
            >
              global.optional
               
              <OverlayTrigger
                defaultOverlayShown={false}
                overlay={
                  <_default
                    className="text-left"
                    id="tooltip-top"
                    style={
                      Object {
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    <FormattedHTMLMessage
                      id="admin.help.metadescription"
                      tagName="span"
                      values={Object {}}
                    />
                  </_default>
                }
                placement="top"
                trigger={
                  Array [
                    "hover",
                    "focus",
                  ]
                }
              >
                <ForwardRef(ProjectContentAdminForm__InformationIcon) />
              </OverlayTrigger>
            </span>
          </div>
        }
        name="metaDescription"
        type="textarea"
      />
    </div>
  </ProjectAdminFormstyle__ProjectBoxContainer>
</div>
`;
